### PR TITLE
[Middleware_02] Define MiddlewareCommLinkManagerImplementationTests

### LINKED ISSUE #70: [Middleware_02] Build and Test Middleware_02 Production Classes
**What is the use case? Please describe.**
For this sprint task, you'll be building and testing all of the classes in the `edu.oakland.production.middleware02` package. Note: This issue is similar to #64 , but is tracked differently as helper and production classes have different deliverable dates.

**Sprint Task Setup**

To be done by @mckenzieking , @kwillard34 , or @rcapollari before the rest of the team starts working on it:

1. Edit this issue message to add a checkbox for each class, and assign it to the correct team member from #59 .

**Prerequisites**

Please makes sure that your test plan from #59 for the class you are writing is closed and merged to main before starting this, to ensure that all of the documentation you need for this is in the correct spot.

**Steps**

To be done by each team member for each class they are developing a test plan for:

1. Each team member should watch the last ~18 minutes of @andrewdimmer 's [Test Driven Development Crash Course](https://youtu.be/eSc0aY4GhPs)
2. One person should create a branch off of main called `define-Name_Of_Class` (ex. the `Demo` classes and tests get build on a branch called `build-Demo`). Everyone else in the programming pair or group should then checkout that branch.
3. Using the information in the video, use test driven development to build all the required classes, stubs, and tests. Remember: you've already defined what you need to build at https://github.com/andrewdimmer/csi-3370-software-project/tree/main/docs/uml-class-diagrams/middleware02 !
4. Commit your changes with the message `[Middleware_02] Built Name_Of_Class`
5. Push your changes
6. Create a pull request with the name `[Middleware_02] Built Name_Of_Class` and the rest of the information like description, tags, etc. For reference, the settings on the right hand side should match this issue, with the exception that the assignees on the PR are just those who are pushing code.
7. Once your code is complete, add @andrewdimmer as a reviewer.

**Tasks to Complete**

- [x] Allocate Team Members to Classes - @mckenzieking , @kwillard34 , or @rcapollari
- [x] Build MiddlewareCommInterfaceImplementation - @trode21 and @NMoua-Edu
- [ ] Build MiddlewareCommLinkManagerImplementation - @tomtomtomcat and @nbergamin123
- [x] Build MiddlewareCommDatabaseInterfaceImplementation - @rcapollari and @GitLee32

**Additional context**
Note: if for some reason you need to change a public interface or shared data type, reach out to @andrewdimmer who can help you manage making that BREAKING CHANGE.

Note: if you run into merge conflicts, let @andrewdimmer know and he can help you resolve them.

### PR DESCRIPTION
**Pull request by kwillard34**
working on #70

### CODE REVIEW

File: docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 package edu.oakland.test.middleware02 {
     class MiddlewareCommLinkManagerImplementationTests {
-
+      + void verifyTrackLengthTest()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml]: Note: you need a test cases for each and every function that your class does. I would take another look at the activity diagram and see if that helps you come up with more of the required test cases.

**Review by andrewdimmer — CHANGES_REQUESTED** [review]
Can we beef this up a bit more? I know this has a lot of your team's logic, so we should make sure that all of the code paths and functional requirements from the customer are covered in tests.

**Review by mckenzieking — APPROVED** [review]

File: docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml [hunk 2]
```diff
@@ -2,6 +2,14 @@
...
 package edu.oakland.test.middleware02 {
     class MiddlewareCommLinkManagerImplementationTests {
+        + void rfidOffSet()
+        + void storeRequestSent()
+        + void emptyTrackData()
+        + void someTrackData()
+        + void enoughTrackData()
+        + void commDatabaseInterfaceIsNotNull()
+        + void gisManagerIsNotNull()
+        + void storedDataCheck()
```
Comment (andrewdimmer) [inline docs/uml-class-diagrams/middleware02/test/MiddlewareCommLinkManagerImplementationTests/MiddlewareCommLinkManagerImplementationTests.puml]: Just a few minor naming correction to better describe the tests, then you're all set to go!
```suggestion
        + void rfidDoesNotMatchStopsTestCase()
        + void storeRequestSent()
        + void emptyDatabaseReturnsEmptyTrackData()
        + void notEnoughDataReturnsPartialTrackData()
        + void enoughTrackDataReturnsFullData()
        + void commDatabaseInterfaceIsNotNull()
        + void gisManagerIsNotNull()
        + void storedTrackDataCheck()
```

**Review by andrewdimmer — CHANGES_REQUESTED** [review]

**Review by mckenzieking — APPROVED** [review]

**Review by andrewdimmer — APPROVED** [review]
Looks good to me, you're all set to merge!

**Pull request merged**